sample/samplers.go

```diff
@@ -87,8 +87,9 @@ func (s *Sampler) sample(tokens []token) (token, error) {
 	// topK also sorts the tokens in descending order of logits
 	tokens = topK(tokens, s.topK)
 
-	tokens = temperature(tokens, s.temperature)
-	tokens = softmax(tokens)
+	// scale and normalize the tokens in place
```

Comment:
This is getting into bike-shed territory, but what do you think of a pattern like this? Its more clear to me that the tokens are being modified in this case.
```
tokens.ApplyTemperature(s.temperature)
tokens.ApplySoftmax()
```

Reply:
A big reason we went from the earlier `Transform` pattern was to keep things a bit more explicit when building the sampling chain especially as it's not interfaced from the runner anymore. I think okay to keep this way for now.